presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java

```diff
@@ -167,6 +157,9 @@ public boolean isFinished()
         if (spilledLookupJoiner.isPresent()) {
             return spilledLookupJoiner.get().isBlocked();
         }
+        if (currentLookupPartition.isPresent()) {
+            return toListenableFuture(currentLookupPartition.get());
```

Comment:
We also have to make sure that when we are logically waiting on two futures to complete, that we return combined future of both of them. If not, we could return almost finished future, that completes before caller checks `isDone` on it.